Additional CLI debug commands (#10450)
solo-io/gloo: docs/content/reference/cli/glooctl_debug.md

---
## glooctl debug

Debug Gloo (requires Gloo running on Kubernetes)

### Synopsis

> Dumps state information to a local directory (or to stdout, for some subcommands). This is useful for debugging failures. If no subcommand is specified, all state information from the `kube`, `gloo`, and `envoy` subcommands will be included. See the subcommand descriptions for more details.

Reviewer: ```suggestion
Dumps state information to a local directory (or to stdout, for some subcommands). This is useful for debugging failures. If no subcommand is specified, this command returns all the state information from the `kube`, `gloo`, and `envoy` subcommands. See the subcommand descriptions for more details.
```

### Options

```

>   -d, --directory string         directory to write debug info to (default "debug")

Reviewer: We mentioned that stdout was also an option...if they wanted stdout for the subcommands that support it, would they have to put that in this flag? or does it just only do stdout instead of writing to a directory for the ones that do stdout?

Author: Yeah, the ones that do stdout just do stdout and the ones that do a directory just do a directory. The logic here basically is that I'm leaving the previous commands unchanged and the new ones really have too much info to justify writing to stdout. So I was just trying to not mislead with the root cmd's description, but if we think the mention of stdout is confusing, I can take it out.

Author: Also, the `debug kube` command really encompasses the info in `logs` and much of the info in `yaml` but in a more readable/better labeled format (afaict `debug logs` just appends logs from several pods without any distinction), so arguably we should deprecate those commands in favor of the new ones. But I would consider that out of scope.

Reviewer: Ok thanks for the explanation. I just wasn't sure if they'd have to do something extra to get stdout, but sounds like it's the default and then if they want, they can just use the --directory option, so I think it's fine.

For the `debug logs/debug yaml`, I can add it to our doc issue to check if we use those anywhere and just change them to `debug kube`.

### SEE ALSO

* [glooctl](../glooctl)	 - CLI for Gloo

> * [glooctl debug envoy](../glooctl_debug_envoy)	 - Dump envoy admin interface to a local directory (requires Gloo running on Kubernetes)

Reviewer: Would this be for a particular gateway proxy? all gateway proxies? If so, maybe add that
```suggestion
* [glooctl debug envoy](../glooctl_debug_envoy)	 - Dump information from the Envoy admin interface for a gateway proxy to a local directory (requires Gloo running on Kubernetes)
```
